.github/src/changed-files.js

```diff
@@ -0,0 +1,114 @@
+// @ts-check
+
+import { diff } from "./git.js";
+
+/**
+ * @param {import('github-script').AsyncFunctionArguments['core']} core
+ * @param {string} [baseCommitish] Defaults to "HEAD^".
+ * @param {string} [headCommitish] Defaults to "HEAD".
+ * @returns {Promise<string[]>} List of changed files, relative to the repo root.  Example: ["specification/foo/Microsoft.Foo/main.tsp"]
+ */
+export async function getChangedFiles(
+  core,
+  baseCommitish = "HEAD^",
+  headCommitish = "HEAD",
+) {
+  // TODO: If we need to filter based on status, instead of passing an argument to `--diff-filter,
+  // consider using "--name-status" instead of "--name-only", and return an array of objects like
+  // { name: "/foo/baz.js", status: Status.Renamed, previousName: "/foo/bar.js"}.
+  // Then add filter functions to filter based on status.  This is more flexible and lets consumers
+  // filter based on status with a single call to `git diff`.
+  const result = await diff(baseCommitish, headCommitish, core, "--name-only");
+
+  const files = result.trim().split("\n");
+
+  core.info("Changed Files:");
+  for (const file of files) {
+    core.info(`  ${file}`);
+  }
+  core.info("");
+
+  return files;
+}
+
+// Functions suitable for passing to string[].filter(), ordered roughly in order of increasing specificity
+
+/**
+ * @param {string} [file]
+ * @returns {boolean}
+ */
+export function json(file) {
+  // Extension "json" with any case is a valid JSON file
+  return typeof file === "string" && file.toLowerCase().endsWith(".json");
+}
+
+/**
+ * @param {string} [file]
+ * @returns {boolean}
+ */
+export function readme(file) {
```

Comment:
Do we want to add a filter with `fileName` as another input to match the file? Otherwise, another one filter will be needed for `tspconfig.yaml`.

Reply:
We can add this if and when it's needed.  It might make sense to have a filter for tspconfig.yaml specifically, especially if we want to do things like case-insensitive mapping.

Or the caller can just use their own filter:

```
changedFiles.filter(f => path.basename(f).toLower() === "tspconfig.yaml")
```